src/Devtools/Brittany.hs

```diff
@@ -0,0 +1,49 @@
+module Devtools.Brittany
+  ( Brittany.Config
+  , getBrittanyConfig
+  , testTree
+  ) where
+
+import Control.Monad.Trans.Maybe (runMaybeT)
+import Devtools.Prelude
+import Language.Haskell.Brittany (BrittanyError(..))
+import Test.Tasty.MGolden (goldenTest)
+
+import qualified Data.Text.IO              as Text
+import qualified Language.Haskell.Brittany as Brittany
+import qualified Test.Tasty                as Tasty
+
+testTree :: Brittany.Config -> [FilePath] -> Tasty.TestTree
+testTree config = Tasty.testGroup "Brittany" . fmap singleTest
+  where
+    singleTest :: FilePath -> Tasty.TestTree
+    singleTest filePath = goldenTest filePath filePath $ do
+      content <- Text.readFile filePath
+
+      either (fail . show . fmap showBrittanyError) pure
+        =<< Brittany.parsePrintModule config content
+
+showBrittanyError :: BrittanyError -> String
+showBrittanyError = \case
+  ErrorInput         str   -> str
+  ErrorUnusedComment str   -> str
+  LayoutWarning      str   -> str
+  ErrorUnknownNode   str _ -> str
+  ErrorMacroConfig   str _ -> "When parsing inline config: " <> str
+  ErrorOutputCheck         -> "Output is not syntactically valid."
+
+-- | Defaults to using a default config when no config file is supplied
+--
+getBrittanyConfig :: Maybe FilePath  -> IO Brittany.Config
+getBrittanyConfig mFilePath = do
+  mConfig <- runMaybeT $
+    maybe
+      (pure Brittany.staticDefaultConfig)
+      (Brittany.readConfigs mempty . pure)
+      mFilePath
+
+  let errorMsg
+        = "Failed to obtain Brittany Config from provided file"
```

Comment:
Did you mean to use more than 2 space indent.

(Sorry for all the formatting comments, I realize these will be 100% eliminated thanks to this PR, and I am thankful for it. If my brain knows there are no whitespace problems, then I'll be able to immediately drop down into more intense reviews)